source/plugin/permissions.rst

```diff
@@ -2,9 +2,229 @@
+``SubjectCollections``.
+``CommandSources`` such as ``Players`` are ``Subjects`` by default, but there are many other types of ``Subjects`` as
+well. Anything that has permissions is a Subject even if it just delegates the checks to a contained Subject.
+Permissions can be granted or denied to a Subject. If a permission is neither granted nor denied its setting will be
+inherited. See Inheritance.
+Subjects provide methods to check whether it has a certain permission or not.
+Plugins that use this method should only query for the specific permission they want to check. It is the
+PermissionService's task to respect the permission and subject inheritance.
+
+Example
+~~~~~~~
+
+The following example could be used to check whether the Player is allowed to execute the teleport command.
+
+.. code-block:: java
+    
+    public canTeleport(Player subject, World targetWorld) {
+        return subject.hasPermission("myPlugin.command.teleport.execute") 
+                && (subject.getWorld() == targetWorld
+                        || subject.hasPermission("myPlugin.command.teleport." + targetWorld.getName()));
+    }
+
+Inheritance
+~~~~~~~~~~~
+
+If a ``Subject`` got a permission assigned, it will use that value.
+Otherwise it will be inherited from any parent ``Subject``. It does not matter what kind of parent (ex group or player)
+``Subject`` that might be.
+
+If neither the subject itself nor any parent subjects grant or deny a permission then it will be inherited from the
+default ``Subject``. The default subject can be obtained from the ``PermissionService``.
+It is recommended for plugins to define their own permissions to the default's transient ``SubjectData`` during every
+server start-up. This allows the server owner to overwrite the defaults defined by plugins according to his needs using
+the default's persistent ``SubjectData``.
 
 .. note::
 
-    This page has not been filled yet by our Documentation Team. If you feel like you can help, you can do so on `our
-    GitHub repository <https://github.com/spongepowered/spongedocs>`_. Also see the `related GitHub Issue
-    <https://github.com/SpongePowered/SpongeDocs/issues/32>`_ for more information on what is required.
+    Its only specified for the default ``Subject`` that the persistent ``SubjectData`` takes precedence over transient
+    ones.
+
+If neither the Subject, nor any of its parents, nor the default grant or deny a permission, then it is automatically
+denied.
+
+.. note::
+
+    Order of precedence in descending order:
+    
+    * Subject itself
+    * Parent Subjects
+    * Defaults
+        * Persistent
+        * Transient
+    * Deny permission
+
+SubjectCollections
+==================
+
+A container for subjects that can be used obtained a Subject by name.
+These are the default Subject Collections:
+
+* User
+    * Contains all on-line and off-line ``User`` / ``Player`` that exist.
```

Comment:
Not necessarily. The likelihood is that this will only contain online players.

Some implementations may still persist any changes made, or they might just return a transient dummy subject.